datalad/support/annexrepo.py

```diff
@@ -152,6 +156,14 @@ def _run_annex_command(self, annex_cmd, git_options=[], annex_options=[],
         debug = ['--debug'] if lgr.getEffectiveLevel() <= logging.DEBUG else []
         backend = ['--backend=%s' % backend] if backend else []
 
+        if git_options is None:
+            git_options = []
+        if annex_options is None:
+            annex_options = []
+
+        if not self.always_commit:
+            git_options += ['-c', 'annex.alwayscommit=false']
```

Comment:
this introduces a side effect... e.g. what if I passed options from a list which I had in a variable outside -- this would modify it.  Better just copy, e.g
```
git_options = git_options[:] if git_options else []
```
since it is a common patterns -- may be we would like to have a little helper, e.g. ```get_new_list```?